internal/ignition/ignition.go

```diff
@@ -0,0 +1,378 @@
+
+	return nil
+}
+
+func isBMHFile(file *config_31_types.File) bool {
+	return strings.Contains(file.Node.Path, "openshift-cluster-api_hosts")
+}
+
+func isMOTD(file *config_31_types.File) bool {
+	return file.Node.Path == "/etc/motd"
+}
+
+func fileToBMH(file *config_31_types.File) (*bmh_v1alpha1.BareMetalHost, error) {
+	parts := strings.Split(*file.Contents.Source, "base64,")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("could not parse source for file %s", file.Node.Path)
+	}
+	decoded, err := base64.StdEncoding.DecodeString(parts[1])
+	if err != nil {
+		return nil, err
+	}
+
+	bmh := &bmh_v1alpha1.BareMetalHost{}
+	_, _, err = scheme.Codecs.UniversalDeserializer().Decode(decoded, nil, bmh)
+	if err != nil {
+		return nil, err
+	}
+
+	return bmh, nil
+}
+
+// fixMOTDFile is a workaround for a bug in machine-config-operator, where it
+// incorrectly parses igition when a File is configured to append content
+// instead of overwrite. Currently, /etc/motd is the only file involved in
+// provisioning that is configured for appending. This code converts it to
+// overwriting the existing /etc/motd with whatever content had been indended
+// to be appened.
+// https://github.com/openshift/machine-config-operator/issues/2086
+func (g *installerGenerator) fixMOTDFile(file *config_31_types.File) {
+	if file.Contents.Source != nil {
+		// the bug only happens if Source == nil, so no need to take action
+		return
+	}
+	if len(file.Append) == 1 {
+		file.Contents.Source = file.Append[0].Source
+		file.Append = file.Append[:0]
+		return
+	}
+	g.log.Info("could not apply workaround to file /etc/motd for MCO bug. The workaround may no longer be necessary.")
+}
+
+// modifyBMHFile modifies the File contents so that the serialized BareMetalHost
+// includes a status annotation
+func (g *installerGenerator) modifyBMHFile(file *config_31_types.File, bmh *bmh_v1alpha1.BareMetalHost, host *models.Host) error {
+	inventory := models.Inventory{}
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		return err
+	}
+
+	hw := bmh_v1alpha1.HardwareDetails{
+		CPU: bmh_v1alpha1.CPU{
+			Arch:           inventory.CPU.Architecture,
+			Model:          inventory.CPU.ModelName,
+			ClockMegahertz: bmh_v1alpha1.ClockSpeed(inventory.CPU.Frequency),
+			Flags:          inventory.CPU.Flags,
+			Count:          int(inventory.CPU.Count),
+		},
+		Hostname: host.RequestedHostname,
+		NIC:      make([]bmh_v1alpha1.NIC, len(inventory.Interfaces)),
+		Storage:  make([]bmh_v1alpha1.Storage, len(inventory.Disks)),
+	}
+	if inventory.Memory != nil {
+		hw.RAMMebibytes = int(inventory.Memory.PhysicalBytes / 1024 / 1024)
+	}
+	for i, iface := range inventory.Interfaces {
+		hw.NIC[i] = bmh_v1alpha1.NIC{
+			Name:      iface.Name,
+			Model:     iface.Product,
+			MAC:       iface.MacAddress,
+			SpeedGbps: int(iface.SpeedMbps / 1024),
+		}
+		switch {
+		case len(iface.IPV4Addresses) > 0:
+			hw.NIC[i].IP = iface.IPV4Addresses[0]
```

Comment:
Why set  iface.IPV4Addresses[0] for all interfaces?

Reply:
That's how the [logic in the python implementation](https://github.com/openshift/assisted-ignition-generator/blob/7cd2174e2caf869608c868e00ace84e5d85a38c5/utils.py#L43-L50) works.